lib/mayaUsd/ufe/UsdAttributeHolder.cpp

```diff
@@ -188,14 +189,34 @@ Ufe::Value UsdAttributeHolder::getMetadata(const std::string& key) const
         if (PXR_NS::UsdShadeNodeGraph(usdPrim())) {
             if (PXR_NS::UsdShadeInput::IsInput(_usdAttr)) {
                 const PXR_NS::UsdShadeInput input(_usdAttr);
-                const std::string           metadata = input.GetSdrMetadataByKey(tok);
+                std::string                 metadata = input.GetSdrMetadataByKey(tok);
+                if (metadata.empty() && key == "uiname") {
```

Comment:
Maybe we can store this `"uiname"` as a constant somewhere, I see this string used in multiple parts.
Everything else seems ok to me.